### PR TITLE
Fixed error caused by getDarkMode in edit page

### LINKED ISSUE #154: error caused by getDarkMode in edit page
https://user-images.githubusercontent.com/45493793/116454832-051c0480-a87e-11eb-8f93-b1be1a508e52.mp4

There's some unexpected behaviour on propose change pages.
@bryan-brancotte Please check, because of which commit this get generated.
Even that theme-button is not showing.

### PR DESCRIPTION
**Pull request by raashika03**
### Checklist

- [X] I indicated which issue (if any) is closed with this PR using [closing keywords](https://help.github.com/articles/closing-issues-using-keywords)
- [X] I only changed lines related to my PR (no bulk reformating)
- [X] I indicated the source and check the license if I re-use code, or I did not re-use code
- [X] I made my best to solve only one issue in this PR, or explain why multi had to be solved at once.

### Issue

fix #154 

### Details
this went unnoticed when I proposed those changes.

### CODE REVIEW

**Comment by raashika03** [comment]
Thank you @bryan-brancotte for this [Removal](https://github.com/edamontology/edam-browser/commit/095b1ec8ad54d3c3147459b8972f6d47275773cc#diff-baeed923d832e4a07645086311048f533c26aeccde256e6df626683a12c38f06). I was also doubtful whether to have a single button for the whole website or let it be on all pages. 
This's seems better now.

**Comment by raashika03** [comment]
That bottom-border colour was needed to be changed, cause we changed class="page-header" to id="page-header" for the front page in #143, so those styling which this edit page was using get changed so added it separately. 
@bryan-brancotte

File: js/edit.js [hunk 1]
```diff
@@ -111,6 +110,7 @@ window.onload = function() {
         fill_form(uri, getUrlParameter('parent'), branch);
     });
     browser.current_branch( getUrlParameter('branch'));
+    getDarkMode();
```
Comment (bryan-brancotte) [inline js/edit.js:113]: Not the solution, indeed it "works" but always check the console, you'll learn more on why it was crashing.

**Review by bryan-brancotte — CHANGES_REQUESTED** [review]
You indeed found an issue, but the PRis partially fixing it. I recommend you start a first PR with the css fix which is sound (I'll merge it right away). And a second in which you will investigate why there is a js issue.

File: js/edit.js (hunk 1 above)
Comment (raashika03) [reply js/edit.js:113]: Ok Thank you for the direction :)

Comment (raashika03) [reply js/edit.js]: @bryan-brancotte Done!

File: js/edit.js [hunk 2]
```diff
@@ -93,7 +93,10 @@ uri = "";
 parent_uri=null;
 
 window.onload = function() {
-    getDarkMode();
+    const currentTheme = localStorage.getItem("theme");
```
Comment (bryan-brancotte) [inline js/edit.js:96]: This code is a duplicate of what is already in getDarkMode ([these lines](https://github.com/edamontology/edam-browser/blob/main/js/utils.js#L71-L74)). Duplicating codes should always be avoided if possible. Here wa can do something else than copy pasting, the goal is to uderstand the issue with getDarkMode and make it work. Also keeping getDarkMode the unique entry point for the dark-theme will help #163

**Review by bryan-brancotte — CHANGES_REQUESTED** [review]

File: js/edit.js (hunk 2 above)
Comment (raashika03) [reply js/edit.js:96]: unique entry point, can you explain a bit? :sweat_smile: 
rest I understood how to do. Ok wait let me do first what I undestood then tell.

Comment (raashika03) [reply js/edit.js:96]: these lines 
`if (localStorage.getItem("theme") == "dark")
        document.body.classList.add("dark-mode");`
might be repeated from getDarkMode but it's just cheking in the local storage if the theme is set to dark.
By unique entry point you might mean to completely remove these code from edit.js and just have it in index.js and anyhow let that selected theme continued over edit page.
did I get it right @bryan-brancotte ?

Earlier I checkd in the console, problem was the button code which was not needed so I removed it.

what can be done now other than adding these two lines.

Comment (bryan-brancotte) [reply js/edit.js:96]: With `unique entry point ` I meant that having `getDarkMode();` which can be called in both main page and edit, and in both page allows to switch to the dark theme is a good thing as once a dev call it, (s)he is sure that the theme is adjusted. Also imagine that in the future we want to store this info differently, we should change two location with the copy past which is error prone.

I recommend you to change nothing in edit.js, and focus on how to handle when [.btn-toggle](https://github.com/edamontology/edam-browser/blob/main/js/utils.js#L70) is not found.

Comment (raashika03) [reply js/edit.js:96]: function getDarkMode() {
    if($(".btn-toggle")[0])  {
        const btn = document.querySelector(".btn-toggle");
        const currentTheme = localStorage.getItem("theme");
        if (currentTheme == "dark") {
            document.body.classList.add("dark-mode");
        }
        btn.addEventListener("click", function () {
        document.body.classList.toggle("dark-mode");

        let theme = "light";
        if (document.body.classList.contains("dark-mode")) {
            theme = "dark";
        }
        localStorage.setItem("theme", theme);
        });
    }
    else if (localStorage.getItem("theme") == "dark") {
            document.body.classList.add("dark-mode");
    }	
}



@bryan-brancotte can you check if this is ok.

**Comment by raashika03** [comment]
Indentation change was required because of addition of if-else statement, so reformatting was needed. Please check @bryan-brancotte

**Review by bryan-brancotte — CHANGES_REQUESTED** [review]
Hi,

You indeed spotted the issue well, the solution is working, but is refactoring a too much.

### Indentation 
There was an incorrect indentation [here](https://github.com/edamontology/edam-browser/blob/main/js/utils.js#L76-L82) and it is still present : when opening a new curly bracket indetation should go on level more.

### Smaller test
Also, and even if your solution works, you could have done it with less refactoring : 
* [Here](https://github.com/edamontology/edam-browser/blob/main/js/utils.js#L70) you get the button
* [Here](https://github.com/edamontology/edam-browser/blob/main/js/utils.js#L75) you add a listener
* You could have simply tested right before if btn was null and then returning : 
```
if (null == btn || undefinfed == btn)
    return;

**Comment by raashika03** [comment]
@bryan-brancotte have a look.

**Comment by raashika03** [comment]
If button isn't present then still we need to check for the current theme for application on edit page. We can't simply return no @bryan-brancotte

**Comment by bryan-brancotte** [comment]
Capture d’écran de 2021-05-03 14-35-56

and after fixing the indentation, and removing a tab
```js
function getDarkMode() {
    const btn = document.querySelector(".btn-toggle");
    const currentTheme = localStorage.getItem("theme");
    if (currentTheme == "dark") {
        document.body.classList.add("dark-mode");
    }
    if (null == btn || undefined == btn)
        return;
    btn.addEventListener("click", function () {
        document.body.classList.toggle("dark-mode");
    
        let theme = "light";
        if (document.body.classList.contains("dark-mode")) {
            theme = "dark";
        }
        localStorage.setItem("theme", theme);
    });
}
```

**Comment by raashika03** [comment]
sorry for complicating things :sweat: now good to merge

**Review by bryan-brancotte — APPROVED** [review]

**Comment by bryan-brancotte** [comment]
> sorry for complicating things  now good to merge

not complicating things, learning things

**Pull request merged**